### PR TITLE
clear filemetadata between runs

### LINKED ISSUE #4145: Cross references to labels on chapters show up as "Appendix"  rather than "Chapter"  when book contains appendices
### Bug description

In a book project, using a label on a title, eg

```markdown
# title {#sec-stuff}
```

will normally result in a reference to "Chapter X" (where X is the number) when you do something like `@sec-stuff`. This worked well in 1.2.

However, somewhere in 1.3, it stopped working properly when a book contains appendices. All references to chapters in this way wind up as "Appendix X" 

For example:

image

The cross references above should be to chapters 1 and 2.

To make it easy to reproduce, please see the latest commits on my quarto-test repo at: https://github.com/fuhrmanator/quarto-test

I'm using Windows 10, and Quarto's version is 1.3.129

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

### PR DESCRIPTION
**Pull request by dragonstyle**
We need to reset the state of individual files between filter passes.

The issue causing this is similar to what we discussed yesterday - the state of whether we have 'entered the appendix' is being set by an early filter pass and then not reset for subsequent filter passes. I confirmed this change fixes the issue, but am curious if you've encountered this before and if so is there a better solution (this feels slightly indirect to me).

Fixes #4145

### CODE REVIEW

**Comment by cscheid** [comment]
I haven't run into this type of state issue yet (I think filemetadata is one of the few places we keep external mutable state in the filter chain)

The place you made the fix seems natural to me, since it'll get merged with the other Pandoc filters in our `combineFilters` utility function anytime we actually use the fileMetadata filter.

With that said, I think the way you're doing this resets the state in a slightly different way than what we had before. Previously, if you had two sequential filters in (eg) quarto-pre that both used fileMetadata, the state would not be reset across those two runs (because the global state would only be reset when going from quarto-pre to user filters to quarto-post). Now, the state will be reset at the end of that particular individual, single filter whenever fileMetadata is "done".

The final thing is that you're doing this in a `Pandoc` call. It's possible that if we use combineFilters to merge fileMetadata with other filters which need fileMetadata info in a `Pandoc` call, but depending on the order that you specified the filters to be combined, fileMetadata will have cleared its state. It's unlikely that we're going to be hit by this, but it's worth keeping in mind.

Other than that, LGTM! I'll let you decide if you want to change anything or merge (I'd be totally fine with merging it)

**Review by cscheid — APPROVED** [review]

**Comment by dragonstyle** [comment]
Per our tuple convo I went ahead and implemented what we discussed and reads much better to me, TY and appreciate a followup whenever you have time.

**Review by cscheid — APPROVED** [review]
yeah, that looks exactly right to me.

**Pull request merged**
